SPEC.md

```diff
@@ -118,6 +118,13 @@ Image Format TODO
 * Define security requirements for a container. In particular is any isolation of users required between containers? What user does each application run under and can this be root (i.e. "real" root in the host).
 * Define how apps are supposed to communicate; can they/do they 'see' each other (a section in the apps perspective would help)?
 
+### Pod Manifest
```

Comment:
For arguments sake let's look at what would happen if we just imported the Kubernetes definition wholesale (I skipped the "Uses of pods" and "Alternatives considered" sections because those are expository rationale for the whole concept and so not directly pertinent to this part of our discussion):

> Pods
> In Kubernetes, rather than individual containers, pods are the smallest deployable units that can be created, scheduled, and managed.
 
The only thing fuzzy here is "container", but that's well-known to be a weasel world and I'll touch on that more below. Perhaps "self-contained application environment" or so would be better - I would actually expect the language to change here in future to clarify this (sadly they don't define "container" in their [glossary](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/glossary.md)!). Otherwise, this captures exactly what we're trying to propose.

> What is a pod?
> 
> A pod (as in a pod of whales or pea pod) correspond to a colocated group of Docker containers with shared volumes. 

I will come back to this as "point number 1".

> A pod models an application-specific "logical host" in a containerized environment. It may contain one or more containers which are relatively tightly coupled -- in a pre-container world, they would have executed on the same physical or virtual host.
 
Again, "containers" is weasly, but the rest is completely accurate.

> Like running containers, pods are considered to be relatively ephemeral rather than durable entities. As discussed in life of a pod, pods are scheduled to nodes and remain there until termination (according to restart policy) or deletion. When a node dies, the pods scheduled to that node are deleted. Specific pods are never rescheduled to new nodes; instead, they must be replaced (see replication controller for more details). (In the future, a higher-level API may support pod migration.)
 
Given the current (unfortunate) appc nomenclature, if you s#pod#appc container# in this paragraph it all checks out perfectly.

> Motivation for pods
> 
> Resource sharing and communication
> 
> Pods facilitate data sharing and communication among their constituents.
 
Ah, finally they omitted the word container! Constituents = instances of application images.

> The containers in the pod all use the same network namespace/IP and port space, and can find and communicate with each other using localhost. Each pod has an IP address in a flat shared networking namespace that has full communication with other physical computers and containers across the network. The hostname is set to the pod's Name for the containers within the pod. More details on networking.
 
Again, s#container#instance-of-ACI# and we're fine

> In addition to defining the containers that run in the pod, the pod specifies a set of shared storage volumes. Volumes enable data to survive container restarts and to be shared among the containers within the pod.

Yup

> In the future, pods will share IPC namespaces, CPU, and memory (LPC2013).
 
So here we are actually _ahead_ of the game since we already encapsulate these ideas as first-class to the spec.

> Management
> 
> Pods also simplify application deployment and management by providing a higher-level abstraction than the raw, low-level container interface. Pods serve as units of deployment and horizontal scaling/replication. 
 
This is exactly what I'm trying to clarify/implement with this PR

> Co-location, fate sharing, coordinated replication, resource sharing, and dependency management are handled automatically.

Yup, with the pod being the first-class deployable unit then all of this is implicit.


Now, coming back to point numero one. This paragraph very specifically refers to _Docker_ containers as the singular unit of which the pod is a collection - and then subsequently throughout the document, "container" is presumably shorthand for specifically this. This is unfortunate because I am not even sure how well a "Docker container" is defined. But let's cheat and look at the Docker website: 

> Docker containers are similar to a directory. A Docker container holds everything that is needed for an application to run. Each container is created from a Docker image. Docker containers can be run, started, stopped, moved, and deleted. Each container is an isolated and secure application platform. Docker containers are the run component of Docker.

OK, so this sounds... exactly like an ACI. There is obviously a bunch of stuff missing here (like embedded annotations describing _how_ the application can run, isolation that applies to the level of the application itself, etc), but all of this exists very much in the appc world.

After Kubernetes gets around to implementing the sharing of other namespaces they allude to above (IPC, etc), they will be even closer to the definition that is codified in the appc spec today.